qsc/plot.py

```diff
@@ -0,0 +1,529 @@
+    '''
+    Plot the modulus of the magnetic field B along a field line with
+    the Boozer toroidal angle varphi acting as a field-line following
+    coordinate
+
+    Args:
+      r (float): near-axis radius r where to create the surface
+      alpha (float): Field-line label
+      phimax (float): Maximum value of the field-line following parameter varphi
+      nphi (int): resolution of the phi grid
+    '''
+    if phimax is None:
+        phimax = 10 * np.pi / abs(self.iota)
+    varphi_array = np.linspace(0, phimax, nphi)
+    _, ax = plt.subplots(1, 1, figsize=(10, 6), dpi=80, facecolor='w', edgecolor='k')
+    plt.xlabel(r'$\varphi$')
+    plt.ylabel(r'$B(\varphi)$')
+    plt.title("r = " + str(r) + ", alpha = " + str(alpha))
+    theta = alpha + self.iota * varphi_array
+    plt.plot(varphi_array, self.B_mag(r, theta, varphi_array, Boozer_toroidal=True))
+    #ax.xaxis.set_major_formatter(tck.FormatStrFormatter('%g $\pi$'))
+    #ax.xaxis.set_major_locator(tck.MultipleLocator(base=phimax*abs(self.iota)/np.pi))
+    plt.tight_layout()
+    plt.show()
+    plt.close()
+
+def B_contour(self, r=0.1, ntheta=100, nphi=120, ncontours=10):
+    '''
+    Plot contours of constant B, with B the modulus of the
+    magnetic field, as a function of Boozer coordinates theta and varphi
+
+    Args:
+      r (float): near-axis radius r where to create the surface
+      ntheta (int): Number of grid points to plot in the Boozer poloidal angle.
+      nphi   (int): Number of grid points to plot in the Boozer toroidal angle.
+      ncontours (int): number of contours to show in the plot
+    '''
+    theta_array = np.linspace(0, 2 * np.pi, ntheta)
+    phi_array = np.linspace(0, 2 * np.pi, nphi)
+    phi_2D, theta_2D = np.meshgrid(phi_array, theta_array)
+    magB_2D = self.B_mag(r, theta_2D, phi_2D, Boozer_toroidal=True)
+    magB_2D.shape = phi_2D.shape
+    fig, ax = plt.subplots(1, 1)
+    contourplot = ax.contourf(phi_2D, theta_2D, magB_2D, ncontours)
+    fig.colorbar(contourplot)
+    ax.set_title('|B| for r=' + str(r))
+    ax.set_xlabel(r'$\varphi$')
+    ax.set_ylabel(r'$\theta$')
+    plt.tight_layout()
+    plt.show()
+    plt.close()
+
+def plot_axis(self, nphi=100, frenet_serret=True, nphi_frenet_serret=80, frenet_serret_factor=0.12, savefig=None):
+    '''
+    Plot axis shape and the Frenet-Serret frame along
+    the axis (optional). If frenet_serret is true, creates
+    a mayavi instance showing the axis and nphi_frenet_serret
+    times 3 vectors, corresponding to the tangent, normal and
+    binormal vectors. If frenet_serret is false, creates a
+    matplotlib instance with only a single axis shape
+    curve shown.
+
+    Args:
+      nphi (int): Number of grid points in the axis shape
+      frenet_serret (bool): True plots the Frenet-Serret frame, False it doesn't
+      nphi_frenet_serret (int): Number of Frenet-Serret vectors to show
+      frenet_serret_factor (float): Size of Frenet-Serret vectors
```

Comment:
What do you think about leaving "serret" out of these three argument names, so the names are not so long?

Reply:
Sure. Fixed!